### PR TITLE
Add logs when e2e tests fail

### PR DESCRIPTION
**Pull request by jakalkhalili**

### CODE REVIEW

File: test/e2e/operator.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
+package e2e
+
+import (
+	"bytes"
```
Comment (pawelprazak) [inline test/e2e/operator.go]: please sort

File: test/e2e/operator.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"io"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"testing"
+)
+
+var logLinesLimit int64 = 15
```
Comment (pawelprazak) [inline test/e2e/operator.go]: please add a comment what this limit actually limits

File: test/e2e/operator.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+func getOperatorPod(t *testing.T, namespace string) *v1.Pod {
+	lo := metav1.ListOptions{
```
Comment (pawelprazak) [inline test/e2e/operator.go]: what is 'lo'

File: test/e2e/operator.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+		LabelSelector: labels.SelectorFromSet(map[string]string{
+			"name": "jenkins-operator",
+		}).String(),
+	}
+	podList, err := framework.Global.KubeClient.CoreV1().Pods(namespace).List(lo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(podList.Items) != 1 {
+		t.Fatalf("Operator pod not found, pod list: %+v", podList)
```
Comment (pawelprazak) [inline test/e2e/operator.go]: "Expected exactly one pod, got: '%+v'"

File: test/e2e/operator.go (hunk 2 above)
Comment (pawelprazak) [reply test/e2e/operator.go]: is is a pod log tail limit?

File: test/e2e/operator.go [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	}
+	logs := buf.String()
+
+	return logs, nil
+}
+
+func failTestAndPrintLogs(t *testing.T, namespace string, err error) {
+	operatorPod := getOperatorPod(t, namespace)
+	logs, logsErr := getOperatorLogs(*operatorPod)
+	require.NoError(t, logsErr)
```
Comment (pawelprazak) [inline test/e2e/operator.go]:
```
if logsErr != nil {
  t.Error("Couldn't get pod logs: " + logsErr)
} else {
  t.Log("Last 10 logs from operator:\n" + logs)
}
t.Fatal(err)
```

File: test/e2e/operator.go [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var (
+	podLogTailLimit int64 = 15
+	// MUST match the labels in the test deployment manifest
```
Comment (tomaszsek) [inline test/e2e/operator.go]: in the deployment manifest: `deploy/operator.yaml`

**Pull request merged**